### ISSUE TITLE
API endpoint /Users

### ISSUE THREAD

**Issue by causefx** [MediaBrowser/Emby#3553]
For some reason `ConnectUserId` is no longer being reported on this endpoint:

Example date returned:

```
	"Name": "causefx",
	"ServerId": "xxxxxxxxxxxxxxxxxxxxxxxxxxxxxx",
	"ConnectUserName": "causefx",
	"ConnectLinkType": "LinkedUser",
	"Id": "LocalIDHERE",
	"HasPassword": true,
	"HasConfiguredPassword": true,
	"HasConfiguredEasyPassword": true,
	"LastLoginDate": "2019-05-09T22:05:26.9870807+00:00",
	"LastActivityDate": "2019-05-09T22:05:33.3143329+00:00",
```

From API Docs:

```
    "Name": "string",
    "ServerId": "string",
    "ServerName": "string",
    "ConnectUserName": "string",
    "ConnectUserId": "string",
    "ConnectLinkType": "LinkedUser",
    "Id": "string",
    "PrimaryImageTag": "string",
    "HasPassword": true,
    "HasConfiguredPassword": true,
    "HasConfiguredEasyPassword": true,
    "EnableAutoLogin": true,
    "LastLoginDate": "2019-05-09T20:03:15.930Z",
    "LastActivityDate": "2019-05-09T20:03:15.930Z",
```

**Comment by LukePulverenti** [comment]
Hi, what would you need this for? This was removed due to it not being utilized by any of our apps.

**Comment by causefx** [comment]
It’s for allowing use of authenticating Emby user accounts with Emby Connect accounts locally.

Our app Organizr allows logging in locally via emby local Api or emby connect api...

we query that endpoint to tie it together with the data we get from Emby connect Api.

**Comment by fma965** [comment]
@LukePulverenti I guess this isn't going to be re-added then? surely the point of an API is for external applications to use, if things are removed from the API without documentation stating this then a API is pointless?

**Comment by cactushydrocodone** [comment]
> 
> 
> @LukePulverenti I guess this isn't going to be re-added then? surely the point of an API is for external applications to use, if things are removed from the API without documentation stating this then a API is pointless?

Emby doesnt support third party applications, the API is for their own Apps to communicate with the server.

**Comment by fma965** [comment]
OK, i mean your API documentation is public but the API isn't for the public.

**Comment by softworkz** [comment]
This is not true - just the opposite. We've taken great effort to properly document our API (https://github.com/MediaBrowser/Emby/wiki) and just recently we even added OpenAPI documentation (http://swagger.emby.media/?staticview=true) allowing interactive tryout.
(not with the static version above, though)

What we don't do is API versioning. Unlike static websites we cannot afford to keep multiple compatibility API version layers as we're advancing quickly and this requires frequent changes to certain parts of the API. Basic things are meant to remain compatible, though.

But as a consequence for consumers of the API, this means that it is inevitable to always perform a version check first (`/System/Info/Public`) and proceed only when the target server's version is verified to work with your application.

**Comment by causefx** [comment]
I totally understand that. So, is there any possibility of adding this back in? If not, I’m cool with disabling this portion of my app....

**Comment by softworkz** [comment]
The primary goal of EmbyConnect is enabling users to find "their" servers when they don't have a static IP. In case of your app, once a user is there, there's no more need for that anyway, I believe.

All I can say at this time is that you're making no mistake by disabling this login method..

**Comment by causefx** [comment]
As far as my app goes, it just creates an SSO using EmbyConnect accts as the backend. 

So “Admins” can use emby kind of like LDAP. allowing users to manage their own passwords etc....

**Comment by softworkz** [comment]
I'm not sure if I understood that correctly. I hope your app is not creating EmbyConnect accounts in an automated way...
For administrative tasks you might better use API Key authentication in case you aren't doing so already.

With regards to user accounts, I would recommend focusing on Emby local accounts (or use Emby's LDAP support if your app is implementing an LDAP server).

**Comment by causefx** [comment]
No, no account creation at all.  Say you the Admin has your Emby instance up. You add me as a user and tie my embyConnect acct to it. 

You can now setup my app and I can now login into my app using my EmbyConnect acct.

**Comment by softworkz** [comment]
Do you need to enter your EmbyConnect password in your app, then?

**Comment by causefx** [comment]
Yes, you are presented with a login screen and the app authenticates using:
```
https://connect.emby.media/service/user/authenticate
```

**Comment by softworkz** [comment]
Ah, yes of course, that's the way it is done there. Apologies, I mixed this up with something else.

I think, with native authentication, the process would be just slightly different:

I - as the Admin - would add you as a new user with a random password and inform you about the credentials, asking you to change the password on first login.

**Comment by causefx** [comment]
Correct, I supported both methods.  Emby Local and EmbyConnect.  Nevertheless, I will post a note about this so Users will know what happened.  If Emby does bring back this field (not sure why they wouldn't as I thought it was beneficial) I can easily re-enable it.

If Emby does bring this back, Can someone please post on this thread?

**Comment by Tuumke** [comment]
Is the Emby local login still working? Can't seem to get this to work..

**Comment by fma965** [comment]
Yes it does, well it's working in organizr atleast

**Comment by schmittyd** [comment]
> OK, i mean your API documentation is public but the API isn't for the public.

A lot of things the Emby team have done in the last year or two don't make sense, and seem very short-sighted. A once great open source project has become a commercial mess... un-uniform playback apps, restricting or removing of API endpoints without any communication, lack of up to date documentation in some respects, no apt and snap support on Linux.

**Comment by LukePulverenti** [comment]
Actually the reason for the change here is all based on community feedback. We just don't have more information at this point because we aren't ready to announce anything just yet.

But the loud and clear feedback you've gotten is you want less reliance on cloud accounts and more privatized and personal Emby Server. So that is exactly what we are doing, which is gradually scaling back Emby Connect, and moving more towards having everything self-managed within each Emby Server.

**Comment by LukePulverenti** [comment]
If you guys could just hang tight for a little longer on this, we are working on something we think will be better and will not require the Emby Connect login. This will a) eliminate confusion between by the two login methods (local vs Emby Connect), and b) allow it to be self-managed within your Emby Server. All while still solving the original goal of Emby Connect, which is enabling easy remote connections without having to remember your server IP address. We are just trying to avoid announcing too early before it's actually ready, and that is the reason why we need a little bit of time on this. Thanks.

**Comment by causefx** [comment]
Hmmmm.... I'm interested in what will be announced.  Thanks.

**Comment by fma965** [comment]
@LukePulverenti thanks for the update, glad to see you are taking feedback from the community and making changes to make the experience better. cloud authentication is always a issue if the authentication server goes down.

**Comment by angrycuban13** [comment]
Hi,

Is there a new update on this?

**Comment by pXius** [comment]
Also interested. 
The change in API access made it tedious to set up a new user with the Ombi/Emby/Organizr Combo.

**Comment by cj2tech** [comment]
Also interested. And would like to see this come back

**Comment by cj2tech** [comment]
@LukePulverenti Any word on this since a year ago? Or is this kinda just dead and will stay eol?

**Comment by cameronurnes** [comment]
I poked at the Ombi dev a little in his Discord, and now he has fixed his end for Ombi v4:  https://github.com/tidusjar/Ombi/commit/bffa7d33d4113c2c873e9bc3a3e62b7ea61acce4

**Comment by sgtcoder** [comment]
Can we get an email address too for login? It makes it challenging to login with First Name/Last Name.
